src/pages/Signup.jsx

```diff
@@ -0,0 +1,341 @@
+            required
+          />
+          <br />
+          <InputLabel htmlFor="confirm-password-form">Confirm Password</InputLabel>
+          <FilledInput
+            style={textFieldStyle}
+            id="confirm-password-form"
+            inputProps={{
+              'aria-label': 'Confirm Password'
+            }}
+            value={confirmPassword}
+            onChange={(e) => setConfirmPassword(e.target.value)}
+            type={showConfirmPassword ? 'text' : 'password'}
+            endAdornment={
+              <InputAdornment position="end">
+                <IconButton
+                  aria-label="toggle confirm password visibility"
+                  onClick={() => setShowConfirmPassword(!showConfirmPassword)}
+                  edge="end"
+                >
+                  {showConfirmPassword ? <VisibilityOff /> : <Visibility />}
+                </IconButton>
+              </InputAdornment>
+            }
+            minLength="8"
+            required
+          />
+          <br />
+          <Button
+            variant="contained"
+            color="primary"
+            size="large"
+            aria-label="Sign Up For a Pod"
+            type="submit"
+          >
+            Set up your Pod
+          </Button>
+        </form>
+      </Card>
+    </>
+  );
+};
+
+/**
+ * Signup - First screen in the user registration flow.
+ * Allows users to either create a pod, or sign into an existing pod
+ *
+ * @memberof Pages
+ * @name Signup
+ * @returns {React.Component} - A react page
+ */
+const Signup = () => {
+  const [oidcIssuer] = useState(ENV.VITE_SOLID_POD_SERVER || ENV.VITE_SOLID_IDENTITY_PROVIDER);
+  const [searchParams] = useSearchParams();
+  const caseManagerWebId = decodeURIComponent(searchParams.get('webId'));
+  const [caseManagerName, setCaseManagerName] = useState();
+  const { session, login, podUrl } = useSession();
+  const { addContact, isSuccess } = useContactsList();
+
+  const registerAndLogin = async (email, password, confirmPassword) => {
+    await registerPod(
+      {
+        email,
+        password,
+        confirmPassword
+      },
+      oidcIssuer
+    );
+
+    await login({
+      oidcIssuer,
+      redirectUrl: window.location.href,
+      clientName: 'PASS Registration'
+    });
+  };
+
+  const loadProfileInfo = async () => {
+    try {
+      const profile = await getWebIdDataset(caseManagerWebId);
+      const profileThing = getThing(profile, caseManagerWebId);
+      setCaseManagerName(getStringNoLocale(profileThing, FOAF.name));
+    } catch {
+      setCaseManagerName(null);
+    }
+  };
+
+  useEffect(() => {
+    loadProfileInfo();
+  }, []);
+
+  useEffect(() => {
+    if (session.info?.isLoggedIn)
+      initializePod(session.info.webId, podUrl, caseManagerWebId, session.fetch);
+  }, [session.info.isLoggedIn]);
+
+  useEffect(() => {
+    if (isSuccess && caseManagerWebId) {
+      const [cmFirstName, cmLastName] = caseManagerName?.split(' ') ?? [null, null];
```

Comment:
Think we could address this in another issue, but for splitting up the name between given and family name, think we could be difficult to simply use split since there could be names like mine with `Ka Hung` being the given name and `Lee` being the family name.

I think a strategy people use to use for my mails would be to concat the given name so that it's `KaHung` and leave the final space alone so it'll be `KaHung Lee` in the mail or `Kahung Lee`. An alternative to this was to add new fields so we could explicitly add family and given names to the profile card.